packages/core/src/fil/keychains.ts

```diff
@@ -10,35 +10,100 @@ import sjcl from "../eth/eth-core-lib/sjcl";
 import { PasswordInvalidError } from "../error";
 import crypto from "crypto";
 import { bytesToWord } from "../eth";
+import { FilAccountKey } from "./abstractWallet";
 
-const base32 = require("base32");
+const base32Encode = require("base32-encode");
 const elliptic = require("elliptic");
 const secp256k1 = new elliptic.ec("secp256k1");
+const bip32 = require("bip32");
+const bip39 = require("bip39");
 
-export class FilKeychains implements Keychains {
-  changePassword(key: Key, password: string, newPassword: string): KeyWithPriv {
-    const priv = this.decrypt(key, password);
+export interface FilKeyWithPriv extends KeyWithPriv {
+  chainCode: string;
+}
 
-    const ecKey = secp256k1.keyFromPrivate(Buffer.from(priv.slice(2), "hex"));
+export class FilKeychains implements Keychains {
+  changePassword(
+    key: FilAccountKey,
+    password: string,
+    newPassword: string
+  ): FilKeyWithPriv {
+    const seed = this.decrypt(key, password);
+    const priv = bip32.fromSeed(Buffer.from(seed, "hex")).privateKey;
+    const ecKey = secp256k1.keyFromPrivate(priv);
+    const privateKey = `0x${ecKey.getPrivate("hex")}`;
     const publicKey = `0x${ecKey.getPublic(false, "hex").slice(2)}`;
     const address = this.getAddress(publicKey);
-    const newKeyFile = this.encryptPrivToKeyFile(priv, newPassword);
+    const newKeyFile = this.encryptValueToKeyFile(
+      priv.toString("hex"),
+      newPassword
+    );
 
     return {
       address,
       pub: publicKey,
-      priv,
+      priv: privateKey,
       keyFile: newKeyFile,
+      chainCode: key.chainCode,
     };
   }
 
   create(password: string): KeyWithPriv {
-    const ecKeyPair = secp256k1.genKeyPair();
+    const seed = this.generateRandomSeed();
+    const privBuffer = bip32.fromSeed(seed).privateKey;
+    const keyPair = secp256k1.keyFromPrivate(privBuffer);
+    const privateKey = `0x${keyPair.getPrivate("hex")}`;
+    const publicKey = `0x${keyPair.getPublic(false, "hex").slice(2)}`;
```

Comment:
Does `keyPair.getPublic(false, "hex")` start with `0x`? Then `slice(2)` might be redundant.

plus) Doesn't `keyPair.getPrivate("hex")` start with `0x`?

Reply:
`keyPair.getXXX` returns buffer, not string.

Especially, `keyPair.getPublic()` returns 65 bytes with prefix. (it might be `0x04`)
But, our public key does not contain a prefix, so I've removed the prefix from that.

In Ethereum, the same logic has been written.